lib/client.js

```diff
@@ -2396,6 +2396,16 @@ MatrixClient.prototype.getOrCreateFilter = function(filterName, filter) {
             // debuglog("Existing filter ID %s: %s; new filter: %s",
             //          filterId, JSON.stringify(oldDef), JSON.stringify(newDef));
             return;
+        }, function(error) {
+            // {errcode: "M_UNKNOWN", name: "M_UNKNOWN", message: "No row found", data: Object, httpStatus: 404}
+            if (error.httpStatus === 404 && (error.errcode === "M_UNKNOWN" || error.errcode === "M_NOT_FOUND")) {
+                // Clear existing filterId if the localStorage value is out of sync with the server
+                self.store.setFilterIdByName(filterName, null);
+                // Return a null value for existingId further down the promise chain
+                return null;
```

Comment:
should probably return undefined, for consistency with the 'out of sync' case and the 'unknown filter name' case.

(though I don't know why the out of sync case has an explicit `return` rather than just letting flow reach the end of the function.)